qiskit_nature/results/protein_folding_tools/protein_decoder.py

```diff
@@ -0,0 +1,107 @@
+        self._main_chain_lenght = len(side_chain_hot_vector)
+
+    def _bitstring2turns(self, bitstring: str) -> List[int]:
+        """
+        Turns a bitstring encoding the shape of the molecule and decodes it.
+
+        Args:
+            bitstring: string containing the encoded information.
+        Returns:
+            A list of integers decoding the bitstring.
+        """
+        bitstring = bitstring[::-1]
+        encoding = {"00": 0, "01": 1, "10": 2, "11": 3}
+        lenght_turns = len(bitstring) // 2
+        return [encoding[bitstring[2 * i : 2 * (i + 1)]] for i in range(lenght_turns)]
+
+    def _split_bitstring(self) -> Tuple[int, int]:
+        """Returns the amount of bits in the compact solution corresponding
+        to each property they encode."""
+        n_qbits_encoding_main_turns = 2 * (self._main_chain_lenght - 3) - (self._fifth_bit)
+        n_qbits_encoding_side_turns = 2 * sum(self._side_chain_hot_vector)
+        return n_qbits_encoding_main_turns, n_qbits_encoding_side_turns
+
+    def get_main_turns(self) -> List[int]:
+        """
+        Returns the list of turns for the molecule corresponding to best_sequence.
+        The first element of the list corresponds to the turn of the second aminoacid in the peptide.
+        Returns:
+                A list of integers representing the sequence of turns on the molecule
+
+        Notes:
+                The bitstring will end in 0010 corresponding to turn1=(01) and turn2=(00)
+                If the second bead doesn't have a side chain the 6th bit
+                can be set to 1 without loss of generality.
+                In that case index (5) will belong to the list of unused qubits
+                The amount of qubits needed to encode the turns will be 2(N-3) - 1
+                if no side chain on second main bead or 2(N-3) otherwise.
+        """
+
+        main_turns_bitstring = self._best_sequence[-self._split_bitstring()[0] :] + "0010"
+
+        if self._fifth_bit:
+            main_turns_bitstring = main_turns_bitstring[:-5] + "1" + main_turns_bitstring[-5:]
+
+        return self._bitstring2turns(main_turns_bitstring)
+
+    def get_side_turns(self) -> List[Union[None, int]]:
+        """
+        Returns the list of turns from the main bead corresponding to the side chains.
+        None corresponds to no side chain from that main bead.
+
+        Returns:
+            A list with either an number associated to a turn from
+            the main bead or None if there is no side bead.
+
+        """
+        n, m = self._split_bitstring()
+
+        side_turns_bitstring = self._best_sequence[-n - m : -n]
+
+        side_turns = self._bitstring2turns(side_turns_bitstring)
+
+        result = []
+        counter = 0
+        for element in self._side_chain_hot_vector:
+            if element:
+                result += [side_turns[counter]]
+                counter += 1
+            else:
+                result += [None]
```

Comment:
```suggestion
                result.append(side_turns[counter])
                counter += 1
            else:
                result.append(None)
```
Do you think this might be cleaner?